### PR TITLE
feat(pipes): set pipe registry during hydration

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: .settings/settings.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
 		".idea": true,
 		"node_modules" : true,
 		"bower_components" : true,
+		"docs": true,
```
Comment (rkirov) [inline .settings/settings.json]: remove

File: modules/angular2/src/core/compiler/element_injector.ts [hunk 2]
```diff
@@ -54,27 +54,28 @@ import {
...
+export class StaticKeys {
   viewManagerId: number;
   protoViewId: number;
   viewContainerId: number;
   changeDetectorRefId: number;
   elementRefId: number;
+  PipeRegistryKey: Key;
 
   constructor() {
-    // TODO: vsavkin Key.annotate(Key.get(AppView), 'static')
```
Comment (rkirov) [inline modules/angular2/src/core/compiler/element_injector.ts]: Add a TODO to change all to ids and explanation why using ids is better - can be used by public API.

File: modules/angular2/src/core/compiler/element_injector.ts [hunk 3]
```diff
@@ -445,6 +446,23 @@ export class ElementInjector extends TreeNode<ElementInjector> implements Depend
   private _injector: Injector;
   private _strategy: _ElementInjectorStrategy;
 
+  static getPipeRegistry(imperativelyCreatedInjector: Injector,
+                         host: ElementInjector): PipeRegistry {
+    var PipeRegistryKey = StaticKeys.instance().PipeRegistryKey;
+    var inj = ElementInjector.parentInjector(imperativelyCreatedInjector, host);
+    return isBlank(inj) ? null : inj.getOptional(PipeRegistryKey);
```
Comment (rkirov) [inline modules/angular2/src/core/compiler/element_injector.ts]: Other than tests, can it happen that we return null here?

File: modules/angular2/src/core/compiler/element_injector.ts [hunk 4]
```diff
@@ -445,6 +446,23 @@ export class ElementInjector extends TreeNode<ElementInjector> implements Depend
...
+  }
+
+  static parentInjector(imperativelyCreatedInjector: Injector, host: ElementInjector): Injector {
```
Comment (rkirov) [inline modules/angular2/src/core/compiler/element_injector.ts]: parentInjector + host in the args is really confusing. I would name it in a way that describes what it does, not how it is being used in ViewManager.

File: modules/angular2/src/core/compiler/element_injector.ts [hunk 5]
```diff
@@ -445,6 +446,23 @@ export class ElementInjector extends TreeNode<ElementInjector> implements Depend
   private _injector: Injector;
   private _strategy: _ElementInjectorStrategy;
 
+  static getPipeRegistry(imperativelyCreatedInjector: Injector,
+                         host: ElementInjector): PipeRegistry {
+    var PipeRegistryKey = StaticKeys.instance().PipeRegistryKey;
+    var inj = ElementInjector.parentInjector(imperativelyCreatedInjector, host);
```
Comment (rkirov) [inline modules/angular2/src/core/compiler/element_injector.ts]: This feels too specific to the usecase in view manager. I think `getPipeRegistry` should just do the lookup from a single Injector and the user (view manager) should call `ElementInjector.parentInjector` separately.

File: modules/angular2/tsconfig.json [hunk 6]
```diff
@@ -1,6 +1,7 @@
 {
     "compilerOptions": {
         "emitDecoratorMetadata": true,
+        "experimentalDecorators": true,
```
Comment (rkirov) [inline modules/angular2/tsconfig.json]: What does this do?

File: .settings/settings.json (hunk 1 above)
Comment (vsavkin) [inline .settings/settings.json]: done

File: modules/angular2/src/core/compiler/element_injector.ts (hunk 2 above)
Comment (vsavkin) [inline modules/angular2/src/core/compiler/element_injector.ts]: done

File: modules/angular2/src/core/compiler/element_injector.ts (hunk 4 above)
Comment (vsavkin) [inline modules/angular2/src/core/compiler/element_injector.ts]: Changed it around. Inline getPipeRegistry into ViewManagerUtils. And renamed parentInjector into closestBoundaryInjector.

File: modules/angular2/tsconfig.json (hunk 6 above)
Comment (vsavkin) [inline modules/angular2/tsconfig.json]: Required for the new version of typescript and vs code

Comment (vsavkin) [inline modules/angular2/tsconfig.json]: I moved it into a separate commit
